Add support for specifying modifiers in HX-Reswap

xhaggi opened the pull request.

wimdeblauwe commented on htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxReswap.java:
```diff
@@ -0,0 +1,279 @@
+     * @param enabled {@code true} if auto-scrolling to focused inputs between requests is enabled, otherwise {@code false}
+     * @return self
+     */
+    public HtmxReswap focusScroll(boolean enabled) {
```
The order of the methods in the class could be a bit more consistent:

- I propose to put equals, hashcode and toString method together
- Move this `focusScroll` builder method with the other builder methods.

xhaggi replied:
I automatically sort the methods by visibility and in alphabetical order. But if you prefer, I can sort them manually.

wimdeblauwe commented on htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxReswap.java:
```diff
@@ -0,0 +1,279 @@
+    }
+
+    @Override
+    public String toString() {
```
Might want to use a `StringBuilder` here instead of all the string concatenation.

xhaggi replied:
Sure.

xhaggi replied:
BTW IntelliJ is suggesting that this should be replaced by String concatenation. But if you prefer StringBuilder no problem.

wimdeblauwe commented on htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxViewHandlerInterceptor.java:
```diff
@@ -118,7 +118,7 @@ private void addHxHeaders(HtmxResponse htmxResponse, HttpServletResponse respons
         }
         if (htmxResponse.getReswap() != null) {
-            response.setHeader(HtmxResponseHeader.HX_RESWAP.getValue(), htmxResponse.getReswap().getValue());
+            response.setHeader(HtmxResponseHeader.HX_RESWAP.getValue(), htmxResponse.getReswap().toString());
```
I would add a different method to convert the `HtmxReswap` object to the string interpretation that htmx uses in the header. Maybe something like `asHeaderValue()`. Normally `toString()` is for debugging values and we should be free to change that at any time I think.

xhaggi replied:
Ok.

xhaggi commented:
@wimdeblauwe I have addressed all your comments. Furthermore I have fixed the return type of `HtmxResponse#getReswap` which I had overlooked and added one more test.
